specifyweb/businessrules/rules/tree_rules.py

```diff
@@ -17,9 +23,57 @@ def cannot_delete_root_treedefitem(sender, obj):
                  "node": {
                      "id": obj.id
                  }})
+        pre_tree_rank_deletion(sender, obj)
 
+@orm_signal_handler('post_delete')
+def post_tree_rank_deletion_handler(sender, obj):
+    if hasattr(obj, 'treedef'): # is it a treedefitem?
+        post_tree_rank_deletion(obj)
 
 @orm_signal_handler('pre_save')
 def set_is_accepted_if_prefereed(sender, obj):
     if hasattr(obj, 'isaccepted'):
         obj.isaccepted = obj.accepted_id == None
+
+def post_tree_rank_initiation(tree_def_item_model, new_rank):
+    tree_def = new_rank.treedef
+    parent_rank = new_rank.parent
+    new_rank_id = new_rank.rankid
```

Comment:
I think instead of trusting the rank id from the front-end here, that's where you should have your smart logic that determines new rank id
OR, we should move the rank id determination logic to the front-end